src/common/image.c

```diff
@@ -2457,8 +2457,11 @@ void dt_image_write_sidecar_file(const int32_t imgid)
       DT_DEBUG_SQLITE3_BIND_INT(stmt, 1, imgid);
       sqlite3_step(stmt);
       sqlite3_finalize(stmt);
+      return 0;
     }
   }
+
+  return 1; // error : nothing written
```

Comment:
return FALSE;

Reply:
I'm not sure we are talking about the same thing.

`success = dt_history_load_and_apply(entry.id, entry.xmp_path, 0);` is a function that was already there. If everything goes as planned, it returns 0 as `int` so `success == (int)0`.

I made `success = dt_image_write_sidecar_file(entry.id); ` behave the same (same values and same type) to be able to handle the success code uniformingly. Also, this function uses Exiv2 XMP writing methods that return 0 in case of success too. 

Now, I get that having a logical `FALSE` returned when everything behaved as planned is not intuitive, but if we are to change that, we need to change `dt_history_load_and_apply()` too (or handle returns code differently depending on the function, which will open the path to human mistakes). Also, I think it's more or less standard in Unix world to have the proper behaviour return 0.